portalnet/src/overlay_service.rs

```diff
@@ -758,58 +758,146 @@ where
+                        let validator = self.validator.clone();
+                        let store = self.store.clone();
+                        let kbuckets = self.kbuckets.clone();
+                        let command_tx = self.command_tx.clone();
+                        tokio::spawn(async move {
+                            metrics.report_utp_active_inc(UtpDirectionLabel::Inbound);
+                            let mut stream =
+                                match utp.connect_with_cid(cid.clone(), UTP_CONN_CFG).await {
+                                    Ok(stream) => stream,
+                                    Err(err) => {
+                                        metrics.report_utp_outcome(
+                                            UtpDirectionLabel::Inbound,
+                                            UtpOutcomeLabel::FailedConnection,
+                                        );
+                                        warn!(
+                                            %err,
+                                            cid.send,
+                                            cid.recv,
+                                            peer = ?cid.peer.client(),
+                                            "Unable to establish uTP conn based on Accept",
+                                        );
+                                        return;
+                                    }
+                                };
+
+                            let mut data = vec![];
+                            if let Err(err) = stream.read_to_eof(&mut data).await {
+                                metrics.report_utp_outcome(
+                                    UtpDirectionLabel::Inbound,
+                                    UtpOutcomeLabel::FailedDataTx,
+                                );
+                                error!(%err, cid.send, cid.recv, peer = ?cid.peer.client(), "error reading data from uTP stream");
+                                return;
+                            }
 
-                    // If content was found, then offer the content to the closest nodes who did
-                    // not possess the content.
-                    if let Some(content) = content {
-                        self.poke_content(content_key, content, closest_nodes);
+                            // report utp tx as successful, even if we go on to fail to process the payload
+                            metrics.report_utp_outcome(
+                                UtpDirectionLabel::Inbound,
+                                UtpOutcomeLabel::Success,
+                            );
+
+                            let trace = query_info.trace;
+                            Self::process_received_content(
+                                kbuckets,
+                                command_tx,
+                                validator,
+                                store,
+                                (data.clone(), true),
```

Comment:
Can't this be moved instead of cloned? This isn't even just a stylistic question, since this data can be a pretty significant size: cloning it might have a measurable performance impact.